native-token/src/lib.rs

```diff
@@ -4,15 +4,25 @@
 
 /// There are 10^9 lamports in one SOL
 pub const LAMPORTS_PER_SOL: u64 = 1_000_000_000;
+const LAMPORTS_PER_SOL_F64: f64 = LAMPORTS_PER_SOL as f64;
 
 /// Approximately convert fractional native tokens (lamports) into native tokens (SOL)
 pub fn lamports_to_sol(lamports: u64) -> f64 {
-    lamports as f64 / LAMPORTS_PER_SOL as f64
+    lamports as f64 / LAMPORTS_PER_SOL_F64
 }
 
 /// Approximately convert native tokens (SOL) into fractional native tokens (lamports)
 pub fn sol_to_lamports(sol: f64) -> u64 {
-    (sol * LAMPORTS_PER_SOL as f64) as u64
+    const U64_MAX_PLUS_1: f64 = 18446744073709551616.; // 0x1p64
+    let lamports = (sol * LAMPORTS_PER_SOL_F64).round();
```

Comment:
Typical practice is to floor() so you don't mint/send more than you ask for (and maybe erroring if you're thus exceeding available balance), any reason to choose not to floor @svenski123 ?

Reply:
Got it. I'm used to absolute under-mint safety, but in this context your reasoning makes more sense.